format/spec.md

```diff
@@ -673,7 +673,7 @@ For serialization details, see Appendix C.
 
 #### Table statistics
 
-Table statistics files are valid [Puffin files](../puffin-spec). Statistics are informational. A reader can choose to
+Table statistics files are valid [Puffin files](puffin-spec.md). Statistics are informational. A reader can choose to
```

Comment:
The website address is https://iceberg.apache.org/puffin-spec/ which is nice.

Will it now be https://iceberg.apache.org/puffin-spec.md (with kind of useless ".md")?
do we plan to also move https://iceberg.apache.org/spec/ to https://iceberg.apache.org/spec.md?

Reply:
I have tagged @samredai for website-related doubts. 

I raised this PR because when I browse a `spec.md` file (not from the website) puffin link was broken as it was looking puffin-spec in the parent directory. 

I want to have links that work in both website and offline markdown file.